ldms/src/ldmsd/ldmsd_request.c

```diff
@@ -6742,3 +6763,80 @@ static int set_default_authz_handler(ldmsd_req_ctxt_t reqc)
 
 	return 0;
 }
+
+extern struct option long_opts[];
+extern char *short_opts;
+extern int ldmsd_process_cmd_line_arg(char opt, char *value);
+static int __cmdline_options(ldmsd_req_ctxt_t reqc, int argc, char *argv[])
+{
+	int opt, opt_idx;
+	reqc->errcode = 0;
+	opterr = 0;
+	optind = 0;
+	while (0 < (opt = getopt_long(argc, argv,
+				  short_opts, long_opts,
+				  &opt_idx))) {
+		switch (opt) {
+		case 'x':
+			reqc->errcode = EINVAL;
+			snprintf(reqc->line_buf, reqc->line_len,
+					"The 'option' command does not support 'x' or 'xprt'. "
+					"Use the 'listen' command instead.");
+			return reqc->errcode;
+		case 'F':
+		case 'u':
+		case 'V':
+			reqc->errcode = EINVAL;
+			snprintf(reqc->line_buf, reqc->line_len,
+					"The option '%s' must be given at the command line.",
+					argv[optind-1]);
+			return reqc->errcode;
+		default:
+			reqc->errcode = ldmsd_process_cmd_line_arg(opt, optarg);
+			if (reqc->errcode == ENOENT) {
+				snprintf(reqc->line_buf, reqc->line_len,
+					"Unknown cmd-line option or it must be "
+					"given at the command line: %s\n", argv[optind-1]);
+				return reqc->errcode;
+			}
+			break;
+		}
+	}
+	return 0;
+}
+
+static int cmd_line_arg_set_handler(ldmsd_req_ctxt_t reqc)
+{
+	char *s, *token, *ptr1;
+	int rc = 0;
+	int argc = 1;
+	int max_opt = 11;
```

Comment:
First of all, this should be marked "const", or be a #define.

By my count, there are 17 short opts. Even subtracting the 4 unallowed short options mentioned in __cmdline_options, that still leaves 13 "supported" options.

Next, this type of hard-coded value has a tendency to cause problems in the future. At the very least, you would also need to add a comment near the definition of short_opts and long_opts in ldmsd.c saying something like "If you add any command line options here, you will need to increment the hard-coded max_opt in ldmsd_request.c:cmd_line_arg_set_handler()".

Better yet would probably be to just write a function that walks both short_opts and long_opts, and returns the total number of options.